helper/helper.go

```
@@ -186,28 +197,78 @@ func getPipeReader() *bufio.Reader {
 }
 
 func getPipeWriter(currentPipe string) (*bufio.Writer, *os.File) {
-	log(fmt.Sprintf("Opening pipe for oid %d", oid))
-	writeHandle, err := os.OpenFile(currentPipe, os.O_WRONLY, os.ModeNamedPipe)
+	fileHandle, err := os.OpenFile(currentPipe, os.O_WRONLY, os.ModeNamedPipe)
 	utils.CheckError(err)
-	writer := bufio.NewWriter(writeHandle)
-	return writer, writeHandle
+	pipeWriter := bufio.NewWriter(fileHandle)
+	return pipeWriter, fileHandle
 }
 
-func closePipe(writeHandle *os.File) {
-	log(fmt.Sprintf("Closing pipe for oid %d", oid))
-	err := writeHandle.Close()
-	utils.CheckError(err)
+func flushAndCloseWriter() {
+	if writer != nil {
+		err := writer.Flush()
+		utils.CheckError(err)
+		writer = nil
+	}
+	if writeHandle != nil {
+		err := writeHandle.Close()
+		utils.CheckError(err)
+		writeHandle = nil
+	}
```

I don't expect writer/writeHandle to ever be nil in this case, during normal operation, but if it were, we'd just be ignoring the error. Before, when we assumed it wasn't nil, it would seg fault (which is better than failing silently).

---

In the termination case, Cleanup might fire twice so setting to nil here and checking at the top prevents the second run from trying to close the file again. We wouldn't expect it to be nil in otherwise normal operation.